frontend/src/components/ApiKeyPageContent.tsx

```diff
@@ -0,0 +1,444 @@
+'use client'
+
+import { FetchResult, useMutation, useQuery } from '@apollo/client'
+import {
+  faSpinner,
+  faKey,
+  faPlus,
+  faTrash,
+  faCopy,
+  faEye,
+  faEyeSlash,
+} from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { Button } from '@heroui/button'
+import { Modal, ModalContent, ModalHeader, ModalBody, ModalFooter } from '@heroui/modal'
+import { Input } from '@heroui/react'
+import { addToast } from '@heroui/toast'
+import { format } from 'date-fns/format'
+import { useSession } from 'next-auth/react'
+import { useEffect, useState, type FC } from 'react'
+import { CREATE_API_KEY, GET_API_KEYS, REVOKE_API_KEY } from 'server/queries/apiKeyQueries'
+import { ApiKey, ApiKeyPageContentProps, CreateApiKeyResult } from 'types/apikey'
+import { userAuthStatus } from 'utils/constants'
+import apolloClient from 'utils/helpers/apolloClient'
+import SecondaryCard from 'components/SecondaryCard'
+
+const ApiKeyPageContent: FC<ApiKeyPageContentProps> = ({ isGitHubAuthEnabled }) => {
+  const { status } = useSession()
+  const [isCreateModalOpen, setIsCreateModalOpen] = useState(false)
+  const [newKeyName, setNewKeyName] = useState('')
+  const [newKeyExpiry, setNewKeyExpiry] = useState('')
+  const [showNewKey, setShowNewKey] = useState(false)
+  const [newlyCreatedKey, setNewlyCreatedKey] = useState<string | null>(null)
+  const [includeRevoked, setIncludeRevoked] = useState(false)
+  const [confirmationModal, setConfirmationModal] = useState<{
+    isOpen: boolean
+    title: string
+    message: string
+    onConfirm: () => Promise<FetchResult<unknown>>
+  } | null>(null)
+
+  const { loading, error, data, refetch } = useQuery(GET_API_KEYS, {
+    variables: { includeRevoked },
+    skip: status !== userAuthStatus.AUTHENTICATED,
```

Comment:
Why should we take care of it on this level? If user is not authenticated there should be no content rendered at all.